File: 2126-destroying-asteroids/2126-destroying-asteroids.java
class Solution {
    public boolean asteroidsDestroyed(int mass, int[] asteroids) {
        // 소행성 크기순으로 오름차순 정렬
        Arrays.sort(asteroids);
        
        long currentMass = mass;
        
        for (int asteroid : asteroids) {
            // 행성의 질량이 소행성보다 작으면 파괴 불가
            if (currentMass < asteroid) {
                return false;
            }
            // 소행성을 파괴하고 질량 추가
            currentMass += asteroid;
        }
        
        return true;
    }
}

File: 2126-destroying-asteroids/2126-destroying-asteroids.ts
function asteroidsDestroyed(mass: number, asteroids: number[]): boolean {
    // 소행성을 오름차순으로 정렬합니다
    asteroids.sort((a, b) => a - b);
    
    let currentMass: number = mass;

    for (const asteroid of asteroids) {
        // 행성의 현재 질량이 소행성의 질량보다 작으면 소행성을 파괴할 수 없습니다
        if (currentMass < asteroid) {
            return false;
        }
        // 소행성을 파괴하고 그 질량을 행성에 더합니다
        currentMass += asteroid;
    }

    return true;
}
